core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/TokenFactory.java

```diff
@@ -15,26 +15,23 @@
  */
 package com.datastax.oss.driver.internal.core.metadata.token;
 
+import com.datastax.oss.driver.api.core.metadata.token.Partitioner;
 import com.datastax.oss.driver.api.core.metadata.token.Token;
 import com.datastax.oss.driver.api.core.metadata.token.TokenRange;
-import java.nio.ByteBuffer;
 
 /** Manages token instances for a partitioner implementation. */
-public interface TokenFactory {
```

Comment:
I don't understand why this change is needed. Why can't TokenFactory by an interface that extends Partitioner interface?

Reply:
Even better would be for TokenFactory.getPartitionerName to be converted into TokenFactory.getPartitioner that returns Partitioner. I'm not sure whether it would fit well with the code though.